src/client.rs

```diff
@@ -141,6 +142,18 @@ impl<IO> AsyncWrite for TlsStream<IO>
 where
     IO: AsyncRead + AsyncWrite + Unpin,
 {
+    fn poll_write_vectored(self: Pin<&mut Self>, cx: &mut Context<'_>, bufs: &[IoSlice<'_>]) -> Poll<Result<usize, Error>> {
```

Comment:
Also process early_data and provide tests.

Reply:
Can you give me some guidance on how to best write tests for this? There are quite a few edge cases that would need testing and I am unsure how to do that.

Reply:
You can refer to https://github.com/rustls/tokio-rustls/blob/main/tests/early-data.rs .
because early rustls does not support server-side early_data, only limited testing can be done with openssl. now we should be able to use rustls for testing.